dpctl/_sycl_timer.py

```diff
@@ -73,25 +111,36 @@ class SyclTimer:
             wall_dt, device_dt = timer.dt
 
     .. note::
-        The timer submits barriers to the queue at the entrance and the
+        The timer submits tasks to the queue at the entrance and the
         exit of the context and uses profiling information from events
         associated with these submissions to perform the timing. Thus
         :class:`dpctl.SyclTimer` requires the queue with ``"enable_profiling"``
         property. In order to be able to collect the profiling information,
         the ``dt`` property ensures that both submitted barriers complete their
         execution and thus effectively synchronizes the queue.
 
+        `device_timer` keyword argument controls the type of tasks submitted.
+        With `device_timer="queue_barrier"`, queue barrier tasks are used. With
+        `device_timer="order_manager"`, a single empty body task is inserted
+        instead relying on order manager (used by `dpctl.tensor` operations) to
+        order these tasks so that they fence operations performed within
+        timer's context.
```

Comment:
It may be nice to add some of the details from this PR description, i.e., 
"Tasks will follow an order of `[prior_tasks] -> [fence_start_task] -> [compute_tasks] -> [fence_end_task] -> [subsequent_tasks]` for some `prior_tasks` started before timing, `compute_tasks` which are being timed, `subsequent_tasks` which are performed after timing, and `fence_start_task` and `fence_end_task` used to find the delta time."

I think this is pretty insightful on its own, though it may belong in documentation or examples rather than this docstring.

Reply:
It may also be good to specify the circumstances under which users will want to use `order_manager` vs. `queue_barrier`, and mention the improved accuracy of `order_manager` implementation.

Reply:
I have pushed changes to the `SyclTimer.__doc__` string.